src/JSInterop/Microsoft.JSInterop/src/DotNetDispatcher.cs

```diff
@@ -23,15 +24,32 @@ public static class DotNetDispatcher
         private static readonly ConcurrentDictionary<AssemblyKey, IReadOnlyDictionary<string, (MethodInfo, Type[])>> _cachedMethodsByAssembly
             = new ConcurrentDictionary<AssemblyKey, IReadOnlyDictionary<string, (MethodInfo, Type[])>>();
 
+        private static readonly AsyncLocal<JSRuntimeBase> _currentJSRuntime = new AsyncLocal<JSRuntimeBase>();
```

Comment:
Wanted to see what it would look like to remove JSRuntime.Current. We only have one piece of code that currently references it (`ReleaseDotNetObject`) which we want to avoid using interop for, so that should clean this up well.

WebAssemblyJSRuntime ends up using a static for it's statically exposed APIs, but that seems reasonable .

Blazor's E2E tests pass, so I don't think I've missed anything. It does involve an API change, perhaps it might be something we want to consider right away rather than sit on it. Thoughts?


-------------------------------

On a related note, I also had a look at trying to removing references to JSInterop & System.Text.Json from Microsoft.AspNetCore.Components. The former is doable today, but the latter is blocked by https://github.com/aspnet/AspNetCore/blob/master/src/Components/Components/src/ElementReference.cs#L15. With the new `JsonSerializerOptions` property we have the ability to have the serialization behavior definied by the runtime, but we'd have to make the ctor and the `Id` property getter public for the server project to be able to read \ write it correctly.

Reply:
How does this halp? This is still an async local 😬

Reply:
We noted this in the API Review, but the plan is to make ReleaseDotNetObject explicit like the other JSRuntime operations \ event handling changes rather than rely on JSInterop for it. I just didn't want to bother with that change as part of this investigation. Once we address that, we would no longer require this.